spec/controllers/users_controller_spec.rb

```diff
@@ -20,77 +21,41 @@
-      end
-      it "doesn't create a new user" do
-        expect(User.count).to eq(0)
-      end
-      it "renders :new" do
-        expect(response).to render_template(:new)
-      end
-      it "sets @user" do
-        expect(assigns(:user)).to be_instance_of(User)
-      end
-    end
-    context "sends emails" do
-      after(:each) { ActionMailer::Base.deliveries.clear }
-      it "sends an email" do
-        post :create, user: Fabricate.attributes_for(:user)        
-        expect(ActionMailer::Base.deliveries).not_to be_empty
-      end
-      it "sends the email to the correct user" do
-        post :create, user: Fabricate.attributes_for(:user)        
-        message = ActionMailer::Base.deliveries.last
-        expect(message.to).to eq([User.first.email])
-      end
-      it "sends the correct content" do
-        post :create, user: Fabricate.attributes_for(:user)        
-        message = ActionMailer::Base.deliveries.last
-        expect(message).to have_content("Thanks for signing up to MyFlix")  
-      end
-    end
-    context "gets invited by a user" do
-      it "creates a relationship where the invitor follows the friend" do
-        invitor = Fabricate(:user)
-        invite = Fabricate(:invite, user: invitor, token: 1234, friend_email: "friend@friend.com")
-        post :create, user: Fabricate.attributes_for(:user, token: 1234, email: invite.friend_email)
-        expect(User.first.following_relationships.first.leader).to eq(User.second)        
-      end
-      it "creates a relationship where the invited friend follows the invitor" do
-        invitor = Fabricate(:user)
-        invite = Fabricate(:invite, user: invitor, token: 1234, friend_email: "friend@friend.com")
-        post :create, user: Fabricate.attributes_for(:user, token: 1234, email: invite.friend_email)
-        expect(User.first.leading_relationships.first.follower).to eq(User.second)        
+
+    context "unsuccessful signup" do
+
+      it "renders the new template" do
+        result = double(:user_signup_response, successful?: false, error_message: "unsuccesful signup" )
+        UserSignup.any_instance.should_receive(:signup).and_return(result)
+        post :create, user: Fabricate.attributes_for(:user)
+        expect(response).to render_template(:new)        
       end
-      it "sets token to nil after use" do
-        invitor = Fabricate(:user)
-        invite = Fabricate(:invite, user: invitor, token: 1234, friend_email: "friend@friend.com")
-        post :create, user: Fabricate.attributes_for(:user, token: 1234, email: invite.friend_email)
-        expect(Invite.first.token).to eq(nil)
+      it "displays error message" do
+        result = double(:user_signup_response, successful?: false, error_message: "unsuccesful signup" )
+        UserSignup.any_instance.should_receive(:signup).and_return(result)
+        post :create, user: Fabricate.attributes_for(:user)
+        expect(flash[:error]).to eq("unsuccesful signup")      
```

Comment:
If you expect it `be_present` instead, simply changing the message text has no chance of making the test fail.